### PR TITLE
Feature/448526 iceland parser

### PR DESCRIPTION
**Pull request by padget11**
# Pull Request Details

## What this PR does / why we need it:

_A brief description of changes being made_
_Link to the relevant work items: e.g: Relates to ADO Work Item AB#213700 and builds on #3376 (link to ADO Build ID URL)_

## Special notes for your reviewer

_Any specific actions or notes on review?_

## Testing

_Any relevant testing information and pipeline runs._

## Checklist (please delete before completing or setting auto-complete)

- [ ] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [ ] Title pattern should be `{work item number}: {title}`
- [ ] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests
- [ ] If the PR includes a new parser, please ensure the happy & unhappy path test files are linked to [this](https://eaflood.atlassian.net/wiki/spaces/EX/pages/4855562467/Packing+List+analysis#Top-20-Traders) page and the detailed description is added to [this](https://eaflood.atlassian.net/wiki/spaces/EX/pages/4855562467/Packing+List+analysis) page. Thanks

## How does this PR make you feel:

gif

### CODE REVIEW

File: app/routes/ai.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        result = fs.readFileSync(filename);
+      } catch (err) {
+        logger.log_error("app/routes/ai.js", "get() > readFileSync", err);
+      }
+
+      const packingList = await findParser(result, filename);
+
+      if (packingList.parserModel !== parser_model.NOMATCH) {
+        const randomInt = Math.floor(
+          Math.random() * (10000000 - 1 + 1) + 1,
```
Comment (david-pizzi-cg) [inline app/routes/ai.js:26]: can you use the getRandomInt function defined in non-ai route?
SonarCloud does not like the use of Math.random...

File: appConfig/appConfig.yaml [hunk 2]
```diff
@@ -44,3 +44,14 @@
   type: "keyvault"
 - key: TP_QUEUE_ADDRESS
   value: "defra.trade.plp.parsed"
+- key: FORM_RECOGNIZER_ENDPOINT
+  value: "https://adpaipocuksdi.cognitiveservices.azure.com/"
```
Comment (david-pizzi-cg) [inline appConfig/appConfig.yaml:48]: Would this value not be different for preprod and prod?

Comment (padget11) [reply appConfig/appConfig.yaml:48]: good point will change that

**Comment by padget11** [comment]
AB#448526

**Pull request merged**